src/aiida/orm/nodes/data/code/portable.py

```diff
@@ -177,3 +180,33 @@ def filepath_executable(self, value: str) -> None:
             raise ValueError('The `filepath_executable` should not be absolute.')
 
         self.base.attributes.set(self._KEY_ATTRIBUTE_FILEPATH_EXECUTABLE, value)
+
+    def _prepare_yaml(self, *args, **kwargs):
+        """Export code to a YAML file."""
+        try:
+            target = pathlib.Path().cwd() / f'{self.label}'
+            setattr(self, 'filepath_files', str(target))
+            result = super()._prepare_yaml(*args, **kwargs)[0]
+
+            extra_files = {}
+            node_repository = self.base.repository
+
+            # Logic taken from `copy_tree` method of the `Repository` class and adapted to return
+            # the relative file paths and their utf-8 encoded content as `extra_files` dictionary
+            path = '.'
+            for root, dirnames, filenames in node_repository.walk():
+                for filename in filenames:
+                    rel_output_file_path = root.relative_to(path) / filename
+                    full_output_file_path = target / rel_output_file_path
+                    full_output_file_path.parent.mkdir(exist_ok=True, parents=True)
+
+                    extra_files[str(full_output_file_path)] = node_repository.get_object_content(
+                        str(rel_output_file_path), mode='rb'
+                    )
+            _LOGGER.warning(f'Repository files for PortableCode <{self.pk}> dumped to folder `{target}`.')
```

Comment:
Why did you change this implementation in 25f9fc0477916b585879e039859e7053b493cf7b ? Did you actually check that this works? I believe this will raise an exception in `data_export` (that is a mistake in of itself. It seems to always raise if the export function returns the `extra_files` and the error message doesn't make sense, but that seems to have been there for a long time). This is probably why you added your latest commit to catch that exception, but that is just sweeping the problem under the rug.

As it stands, the content of the code's repo is not actually exported. You are writing the content to the `extra_files` dictionary, but that only lives in memory. How is the user supposed to access that once the CLI commands terminates and the memory is cleaned? Surely, this only makes sense when you actually write it to disk as in my original suggestion. The problem is not brought to the surface since you only test `PortableCode._prepare_yaml`, but you don't test the path when going through the CLI and making sure the files are actually written to a directory on disk.

Reply:
Ok correction, I misread and see that the if an output filename is specified then `data_export` calls `Data.export` instead of `Data._exportcontent`, and in `Data.export` I can see that the content of `extra_files` is written to disk. So I guess your implementation does make sense. I do think this export implementation is a bit problematic because it requires reading _all_ of the repository content in memory. This is almost never a good idea, but this is already there and not something your PR is introducing.